frontend/packages/dev-console/src/components/catalog/CatalogController.tsx

```diff
@@ -1,40 +1,158 @@
 import * as React from 'react';
-import { PageHeading, skeletonCatalog, StatusBox } from '@console/internal/components/utils';
+import * as _ from 'lodash';
+import Helmet from 'react-helmet';
+import { useTranslation } from 'react-i18next';
+import {
+  PageHeading,
+  skeletonCatalog,
+  StatusBox,
+  removeQueryArgument,
+  setQueryArgument,
+} from '@console/internal/components/utils';
+import { CatalogItem, CatalogItemAttribute } from '@console/plugin-sdk';
+import { DEV_CATALOG_FILTER_KEY, useQueryParams } from '@console/shared';
+
 import { CatalogService } from './service/CatalogServiceProvider';
+import CatalogView from './catalog-view/CatalogView';
+import useCatalogCategories from './hooks/useCatalogCategories';
+import CatalogDetailsModal from './details/CatalogDetailsModal';
+import CatalogTile from './CatalogTile';
+import { determineAvailableFilters } from './utils/filter-utils';
+import { CatalogFilters, CatalogQueryParams, CatalogStringMap, CatalogType } from './utils/types';
 
 type CatalogControllerProps = CatalogService;
 
 const CatalogController: React.FC<CatalogControllerProps> = ({
   type,
   items,
+  itemsMap,
   loaded,
   loadError,
   catalogExtensions,
 }) => {
-  let title = 'Developer Catalog';
-  let description =
-    'Add shared applications, services, event sources, or source-to-image builders to your project from the Developer catalog. Cluster administrators can customize the content made available in the catalog.';
+  const { t } = useTranslation();
+
+  let title = t('devconsole~Developer Catalog');
+  let description = t(
+    'devconsole~Add shared applications, services, event sources, or source-to-image builders to your project from the developer catalog. Cluster administrators can customize the content made available in the catalog.',
+  );
+
+  const queryParams = useQueryParams();
+  const filterGroups: string[] = [];
+  const filterGroupNameMap: CatalogStringMap = {};
+  const filterPreference: string[] = [];
+  const groupings: CatalogStringMap = {};
+
+  const selectedId = queryParams.get(CatalogQueryParams.SELECTED_ID);
+  queryParams.delete(CatalogQueryParams.TYPE);
+  const breadcrumbs = [
+    {
+      name: t('devconsole~Developer Catalog'),
+      path: `/catalog?${queryParams.toString()}`,
```

Comment:
`path` should include current namespace. Use the path from `useLocation`

Reply:
Done.